nifi-framework-api/src/main/java/org/apache/nifi/authorization/AbstractPolicyBasedAuthorizer.java

```diff
@@ -427,8 +429,7 @@ private PoliciesUsersAndGroups parsePoliciesUsersAndGroups(final String fingerpr
 
         final byte[] fingerprintBytes = fingerprint.getBytes(StandardCharsets.UTF_8);
         try (final ByteArrayInputStream in = new ByteArrayInputStream(fingerprintBytes)) {
-            final DocumentBuilder docBuilder = createSafeDocumentBuilder();
-            final Document document = docBuilder.parse(in);
+            final Document document = parseFingerprint(in);
```

Comment:
Good question! This is only remaining location with a local reference to the `DocumentBuilderFactory`. PR #5514 for NIFI-9069 included the direct implementation, as opposed to `XmlUtils` due to issues with class loading between `nifi-framework-api` and dependent modules. It might be possible to revisit the problem, but this change maintains a limited scope for `nifi-framework-api` module dependencies.